client/src/components/Donation/components/DonationModal.js

```diff
@@ -36,17 +44,54 @@ const mapDispatchToProps = dispatch =>
   );
 
 const propTypes = {
+  activeDonors: PropTypes.number,
   block: PropTypes.string,
   closeDonationModal: PropTypes.func.isRequired,
+  isBlockDonation: PropTypes.bool,
   show: PropTypes.bool
 };
 
 class DonateModal extends Component {
+  state = {
+    isBlockDonation: this.props.isBlockDonation
```

Comment:
I am trying to lock the props value here because after the modal is opened that prop will change its value immediately resulting then modal to change.

Reply:
Thanks for clearing that up, I hadn't realised.

I've pushed a commit that deals with this in a different way, so that the `DonationalModal` can just be a function of it's props, rather than needing to manage its own state.  To an extent this is a matter of taste, I just thought it would be better if we decoupled the display from the creation of the modal.

If you disagree, don't hesitate to change it!